bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/KnxNetFrame.java

```diff
@@ -0,0 +1,204 @@
+     *
+     */
+    public void setStartDataPoint(int value) {
+        this.startDataPoint = value;
+    }
+
+    /**
+     * Sets the value messages of the KNX frame
+     *
+     */
+    public SetDatapointValueMessage[] getValueMessages() {
+        SetDatapointValueMessage[] result = new SetDatapointValueMessage[this.valueMessages.size()];
+        this.valueMessages.toArray(result);
+        return result;
+    }
+
+    /**
+     * Creates a KNX frame based on the data-array
+     *
+     */
+    @Nullable
+    public static KnxNetFrame createKnxNetPackage(byte[] data, int amount) throws Exception {
+        if (data.length < 16 || amount < 16 || data.length < amount) {
+            logger.warn("Length of the data too short for a KNXnet/IP package ({}).", data.length);
+            return null;
+        }
+
+        if (data[0] != knxHeader[0] || data[1] != knxHeader[1] || data[2] != knxHeader[2] || data[3] != knxHeader[3]) {
+            logger.warn("Incorrect KNXnet/IP header.");
+            return null;
+        }
+
+        int frameSize = Byte.toUnsignedInt(data[4]) * 256 + Byte.toUnsignedInt(data[5]);
+        if (frameSize != amount) {
+            logger.warn("CreateKnxNetPackage: Error TelegrammLength/FrameSize missmatch. ({}/{})", data.length,
+                    frameSize);
+            return null;
+        }
+
+        KnxNetFrame frame = new KnxNetFrame();
+        frame.setMainService(data[10]);
+        if (frame.getMainService() != (byte) 0xF0) {
+            logger.warn("CreateKnxNetPackage: Main-Service not supported. ({})", frame.getMainService());
+            return null;
+        }
+
+        if (data[11] == (byte) 0x06) {
+            frame.setSubService(SubServiceType.SET_DATAPOINT_VALUE_REQUEST);
+        } else if (data[11] == (byte) 0x86) {
+            frame.setSubService(SubServiceType.SET_DATAPOINT_VALUE_RESULT);
+        } else if (data[11] == (byte) 0xC1) {
+            frame.setSubService(SubServiceType.DATAPOINT_VALUE_WRITE);
+        } else if (data[11] == (byte) 0xD0) {
+            frame.setSubService(SubServiceType.REQUEST_ALL_DATAPOINTS);
+        } else {
+            logger.warn("CreateKnxNetPackage: Sub-Service not supported. ({})", frame.getSubService());
+            return null;
+        }
+
+        if (frame.getSubService() == SubServiceType.SET_DATAPOINT_VALUE_REQUEST) {
+            frame.setStartDataPoint(Byte.toUnsignedInt(data[12]) * 256 + Byte.toUnsignedInt(data[13]));
+            int numberOfDatapoints = Byte.toUnsignedInt(data[14]) * 256 + Byte.toUnsignedInt(data[15]);
+            int offset = 16;
+
+            ByteBuffer list = ByteBuffer.allocate(data.length);
+            list.put(data);
+
+            for (int i = 0; i < numberOfDatapoints; i++) {
+                byte[] msgData = new byte[amount - offset];
+                for (int j = 0; j < msgData.length; j++) {
+                    msgData[j] = list.get(offset + j);
+                }
```

Comment:
Looks like this could be simplified, too.

Reply:
I changed this into list.postion followed by list.get